AttitudeManager/Inc/AM.hpp

```diff
@@ -9,30 +9,32 @@
 #ifndef ZPSW3_AM_HPP
 #define ZPSW3_AM_HPP
 
-#include "LOS_Link.hpp"
-#include "LOS_Actuators.hpp"
-#include "AM_StateManager.hpp"
-
-class AttitudeState;
+// #include "LOS_Actuators.hpp"
+#include "AM_ControlInterface.hpp"
+#include "AM_DataTypes.hpp"
 
 namespace AM {
-    // Gives status of attitude manager so we know when it has completed a cycle (its state is FetchInstructionsMode) or entered failure mode
-    enum _Attitude_Manager_Cycle_Status {COMPLETED_CYCLE = 0, IN_CYCLE, FAILURE_MODE};
-}
 
 class AttitudeManager {
-public:
-    AttitudeManager(LOS_Link *link, LOS_Actuators *output);
-    inline AttitudeState* getCurrentState() const {return currentState;}
-    void execute();
-    void setState(AttitudeState& newState);
-    AM::_Attitude_Manager_Cycle_Status getStatus() {return status;}
-    LOS_Link *link;
-    LOS_Actuators *output;
-private:
-    AttitudeState* currentState;
-    AM::_Attitude_Manager_Cycle_Status status;
+   public:
+    AttitudeManager(ControlInterfaceList controller_interfaces)
+        : controller_interfaces(controller_interfaces){};
+
+    template <typename... Args>
+    AttitudeManager(Args... controllers)
+        : controller_interfaces{controllers...} {};
```

Comment:
Man, I'm still not used to this kind of syntax, and I don't think many other people are too, maybe put a comment on the technique you are using here so people can search it up?

Reply:
https://en.cppreference.com/w/cpp/language/parameter_pack